storage/backends/s3/s3.go

```diff
@@ -283,3 +347,133 @@ func (b S3backend) GetBasebackup(viper func() map[string]interface{}, backup *ut
 	wgDone.Wait()
 	log.Debug("getFromS3 done")
 }
+
+// DeleteAll deletes all backups in the struct
+func (b S3backend) DeleteAll(backups *backup.Backups) (count int, err error) {
+	// Sort backups
+	backups.SortDesc()
+	// We delete all backups, but start with the oldest just in case
+	for i := len(backups.Backup) - 1; i >= 0; i-- {
+		backup := backups.Backup[i]
+		err = backups.MinioClient.RemoveObject(backup.Path, backup.Name+backup.Extension)
+		if err != nil {
+			log.Warn(err)
+		} else {
+			count++
+		}
+
+	}
+	return count, err
+}
+
+// GetStartWalLocation returns the oldest needed WAL file
+// Every older WAL file is not required to use this backup
+func (b S3backend) GetStartWalLocation(bp *backup.Backup) (startWalLocation string, err error) {
```

Comment:
I am working on this...this is one of the things i would like to change or make easier. Also i think the use of func() map... is not very sexy...